executor/index_lookup_hash_join.go

```diff
@@ -294,7 +420,15 @@ func (iw *indexHashJoinInnerWorker) run(ctx context.Context, cancelFunc context.
 			joinResult.err = task.err
 			break
 		}
-		err := iw.handleTask(ctx, cancelFunc, task, joinResult, h)
+		if task.keepOuterOrder {
+			if resultCh != nil {
+				// resultCh is closed to notice the end of the previous task
+				// when a new task is fetched.
+				close(resultCh)
+			}
+			resultCh = task.resultCh
+		}
+		err := iw.handleTask(ctx, cancelFunc, task, joinResult, h, resultCh)
```

Comment:
If workerA has task1 and task3, workerB has task2
task1 and task3 will share a same joinResult, which breaks the order.